File: design-patterns/creational_pattern/singleton/singleton.js

let instance = null;

class Singleton{
    constructor() {
        if (instance) {
            throw new Error("You can only create one instance!");
        }
        instance = this;
    }

    getInstance(){
        return this;
    }
}

let obj1 = new Singleton()
let obj2 = new Singleton()

obj1.getInstance()
obj2.getInstance()

File: design-patterns/creational_pattern/singleton/singleton.py
class Singleton(object):
    __instance = None

    @staticmethod
    def get_instance():
        if Singleton.__instance is None:
            Singleton()
        return Singleton.__instance

    def __init__(self):
        if Singleton.__instance:
            raise Exception("Sorry! You can only instantiate one time of {0} class.".format(self.__class__.__name__))

        else:
            Singleton.__instance = self


obj1 = Singleton()
obj2 = Singleton()
